### ISSUE TITLE
Support for NGS

### ISSUE THREAD

**Issue by jirenius** [jirenius/resgate#63]
## Issue

Add support to connect Resgate to NGS.

**Comment by autodidaddict** [comment]
It looks like there's a solution to this but this branch hasn't been merged to master. I was about to start doing some experimentation with resgate and noticed that it didn't seem to support NATS 2.0 authentication (a requirement for my production scenario).  

Is this blocked on something @jirenius ?

**Comment by ColinSullivan1** [comment]
Perhaps it was blocked on the official release of NATS 2.0- NGS uses the same mechanism.  The branch looks ready...

BTW, not required, but an example unit test can be found [here](https://github.com/nats-io/nats.go/blob/v1.8.0/nats_test.go#L1388).

**Comment by jirenius** [comment]
This branch wasn't merge because it didn't fully support NGS. One feature was still missing: the ability to know if a message might have been lost within a super-cluster (eg. through a super-cluster node disconnect event).

But since support for authentication credentials is a stand-alone feature, I will have it merged to develop right away.

@autodidaddict , I can mail you new binaries (which architecture?) with authentication support if needed, or post them for you on https://forum.resgate.io .

Thanks @ColinSullivan1 ! That helps when creating the unit tests for it.

**Comment by jirenius** [comment]
Created separate issue #108 for the credentials, and pulled it to `develop` in #109
